contracts/jobs/JobsManager.sol

```diff
@@ -70,94 +163,300 @@ contract JobsManager is IJobsManager, Verifiable, Controllable {
-     * @dev Return job details
+     * @dev Submit transcode receipt and invoke transcoding verification
      * @param _jobId Job identifier
+     * @param _segmentNumber Segment sequence number in stream
+     * @param _dataHash Content-addressed storage hash of segment data
+     * @param _transcodedDataHash Hash of transcoded segment data
+     * @param _broadcasterSig Broadcaster's signature over segment hash
+     * @param _proof Merkle proof for transcode receipt
      */
-    function getJobDetails(uint256 _jobId) public constant returns (uint256, uint256, address, address, uint256) {
-        return jobs.getJobDetails(_jobId);
+    function verify(
+        uint256 _jobId,
+        uint256 _claimId,
+        uint256 _segmentNumber,
+        string _dataHash,
+        string _transcodedDataHash,
+        bytes _broadcasterSig,
+        bytes _proof
+    )
+        external
+        payable
+        returns (bool)
+    {
+        // Must be valid job id
+        require(_jobId < numJobs);
+        // Must be valid claim id
+        require(_claimId < jobs[_jobId].claims.length);
+
+        Job storage job = jobs[_jobId];
+        Claim storage claim = job.claims[_claimId];
+
+        // Job must be active
+        require(jobStatus(_jobId) == JobStatus.Active);
+        // Sender must be elected transcoder
+        require(job.transcoderAddress == msg.sender);
+
+        // Segment must be eligible for verification
+        require(JobLib.shouldVerifySegment(_segmentNumber, claim.segmentRange, claim.claimBlock, verificationRate));
+        // Receipt must be valid
+        require(validateReceipt(_jobId, _claimId, _segmentNumber, _dataHash, _transcodedDataHash, _broadcasterSig, _proof));
+
+        // Mark segment as submitted for verification
+        claim.segmentVerifications[_segmentNumber] = true;
+
+        // Invoke transcoding verification. This is async and will result in a callback to receiveVerification() which is implemented by this contract
+        verifier.verify(_jobId, _claimId, _segmentNumber, _dataHash, _transcodedDataHash, this);
+
+        return true;
     }
 
     /*
-     * @dev Return stream id for job
+     * @dev Callback function that receives the results of transcoding verification
      * @param _jobId Job identifier
+     * @param _segmentNumber Segment being verified for job
+     * @param _result Boolean result of whether verification succeeded or not
      */
-    function getJobStreamId(uint256 _jobId) public constant returns (string) {
-        return jobs.jobs[_jobId].streamId;
+    function receiveVerification(uint256 _jobId, uint256 _claimId, uint256 _segmentNumber, bool _result) onlyVerifier external returns (bool) {
+        if (!_result) {
+            refundBroadcaster(_jobId, _claimId);
+            // Protocol slashes transcoder for failing verification (no finder)
+            bondingManager().slashTranscoder(jobs[_jobId].transcoderAddress, address(0), failedVerificationSlashAmount, 0);
```

Comment:
The tokens are currently just left in the BondingManager contract and not associated with anyone. We could send them to a burn address as well if we want to actually burn them